### DISCUSSION TITLE
How to simulate airplane / offline /throttle network

### DISCUSSION THREAD

**Discussion by nathantaal** [php-webdriver/php-webdriver#1000] [Q&A]
Hello everyone,

Both Firefox and Chromium (based) browsers have throttle options. Is there a way to utilize these tools using php-webdriver. I want to app to fake going online in a test. I see C# and JS selenium having ChromeNetworkConditions, but can seem to find this method inside php-webdriver

**Comment by OndraM** [comment]
Hi, this command is not implemented in php-webdriver.

For reference:
- Chromederiver support: https://bugs.chromium.org/p/chromedriver/issues/detail?id=984
- Ruby implementation: https://github.com/SeleniumHQ/selenium/commit/1103cb7e9e35e8bd22a5aea9b5455445baa6fb85
- Java implementation: https://github.com/SeleniumHQ/selenium/pull/3479/files
- JavaScript implementation: https://github.com/seleniumhq/selenium/commit/d7bdd7cc42faafb1eba031c26b7c369cdea026d5

But I believe this should be easily worked around using our [custom commands](https://github.com/php-webdriver/php-webdriver/wiki/Custom-commands) feature. Something like this: (not tested)

```php
$driver->executeCustomCommand(
    '/session/:sessionId/chromium/network_conditions',
    'POST'
    [
        'offline' => false,
        'latency' => 5, // Additional latency (ms).
        'download_throughput' => 500 * 1024, // Maximal aggregated download throughput.
        'upload_throughput' => 500 * 1024, // Maximal aggregated upload throughput.
    ]
);
```

As for Firefox, it is unfortunately not implemented in Geckodriver, see https://github.com/mozilla/geckodriver/issues/1834 , so it is most probably not possible.
